features/cellular/framework/AT/ATHandler.cpp

```diff
@@ -120,8 +126,15 @@ bool ATHandler::get_debug() const
 
 ATHandler::~ATHandler()
 {
+    ScopedLock <ATHandler> lock(*this);
     set_file_handle(NULL);
 
+    // cancel possible oob event from queue and wait if oob is ongoing
+    _queue.cancel(_event_id);
+    while (_oob_pending) {
```

Comment:
if `AT_HANDLER_MUTEX` was not defined, you can't do the condition variable wait, so I think this would then just be `MBED_ASSERT(!_oob_pending)`. (ie the "cancel" must have worked).

Reply:
If you had separate variables, this would need to be `while (_oob_pending || _oob_queued)`, but I don't think you do need them.

Reply:
As cancel is void function we decided not to call it. 
Replaced _oob_queued  and _oob_pending with _event_id.
Force pushed changes.